core/parentchain/block-importer/src/block_importer.rs

```diff
@@ -176,11 +190,35 @@ impl<
 				},
 				Err(_) => error!("Error executing relevant extrinsics"),
 			};
-			let queue_game: Option<u64> = self
+			let queue_game: Option<Queue<H256>> = self
 				.ocall_api
 				.get_storage_verified(RegistryStorage::queue_game(), block.header())?
 				.into_tuple()
 				.1;
+			match queue_game {
+				Some(mut u) => {
+					//FIXME: hardcoded, because currently hardcoded in the GameRegistry pallet.
+					let game_engine = GameEngine::new(1u8, 1u8);
+					//FIXME: we currently only take the first shard. How we handle sharding in general?
+					let game = u.dequeue();
```

Comment:
If multiple games are queued, we may acknowledge multiple games. As far as I understand, dequeue only pops the first in game in the queue. You should iterate through the full queue I think.

(ack_game expects a vec of games: https://github.com/ajuna-network/ajuna-node/blob/update-substrate-5/pallets/ajuna-gameregistry/src/lib.rs#L315)